frontend/views/top_containers/bulk_operations/_bulk_action_labels.html.erb

```diff
@@ -0,0 +1,90 @@
+<%
+  columns = 4
+  rows = 6
+  column_width = "2.25in"
+  row_height = "4.25in"
+  page_break = "1in" # not implemented
+%>
+
+<html>
+  <head>
+    <title><%= I18n.t("top_container._frontend.bulk_operations.label_page_title") %></title>
+    <link href="<%= "#{AppConfig[:frontend_prefix]}assets/container_labels.css" %>" media="all" rel="stylesheet" type="text/css"/>
+    <script src="<%= "#{AppConfig[:frontend_prefix]}assets/jquery.js?body=1" %>" type="text/javascript"></script>
+    <script src="<%= "#{AppConfig[:frontend_prefix]}assets/jquery-barcode.min.js" %>"></script>
+    <script src="<%= "#{AppConfig[:frontend_prefix]}assets/generate_barcodes.js" %>"></script>
+  </head>
+
+<body>
+  <div id="labels_to_print" class="label-sheet">
+
+    <% columns.times do %>
+      <div class="label-sheet-cell" style="width: <%= column_width %>; height: 0px;"></div>
+    <% end %>
+
+    <% labels.each_slice(columns) do |label_row| %>
+
+      <div class="label-sheet-row">
+        <% label_row.each do |label| %>
+          <% 
+            loc = label['container_locations'].select { |cl| cl['status'] == 'current' }.first
+            loc = loc ? loc['_resolved'] : {}
+            loc = loc.empty? ? '--' : ['building', 'floor', 'room', 'area'].map {|fld| loc[fld]}.compact.join(', ')
+            resource = label['collection'].empty? ? {} : label['collection'].first
+            series = label['series'].empty? ? {} : label['series'].first
+            series = series.empty? ? "" : series['display_string']
+            archival_record = label['linked_records'].empty? ? {} : label['linked_records'].first
+            date = archival_record['dates'].empty? ? {} : archival_record['dates'].first
+            note = archival_record['notes'].empty? ? {} : archival_record['notes'].select{|n| n['type'] == 'scopecontent'}.first
```

Comment:
Hit an error here: `undefined method `empty?' for nil:NilClass`.  You could use `ASUtils.wrap(archival_record['dates']).empty?` to ensure an array is returned.